cpp/tests/io/type_inference_test.cu

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <io/utilities/trie.cuh>
+#include <io/utilities/type_inference.cuh>
+
+#include <cudf_test/base_fixture.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+
+#include <thrust/iterator/zip_iterator.h>
+#include <thrust/tuple.h>
+
+#include <cstddef>
+#include <string>
+#include <vector>
+
+using cudf::io::parse_options;
+using cudf::io::detail::infer_data_type;
+
+// Base test fixture for tests
+struct TypeInference : public cudf::test::BaseFixture {
+};
+
+TEST_F(TypeInference, Basic)
+{
+  auto const stream = rmm::cuda_stream_default;
+
+  auto options       = parse_options{',', '\n', '\"'};
+  options.trie_true  = cudf::detail::create_serialized_trie({"true"}, stream);
+  options.trie_false = cudf::detail::create_serialized_trie({"false"}, stream);
+  options.trie_na    = cudf::detail::create_serialized_trie({"", "null"}, stream);
```

Comment:
does this mean JSON fields with empty input get parsed as `null` fields, or do they already fail at the DFA? I'd assume the latter
cc @elstehle

Reply:
The latter. Currently, this mostly serves as a means of "communicating" ommitted fields.